### ISSUE TITLE
Select attribute doesn't work with mm_tables

### ISSUE THREAD

**Issue by JLBA** [MetaModels/attribute_select#48]
Hi,

I'm trying to add a select attribute in a MM but if I select another MM as source table, Id Column, Value Column and Alias column stay empty. Nothing to select inside… But it works if I select a Contao tl_table.

Under Contao 3.5.6, contao-community-alliance/composer-client 0.16.2, Metamodels/Bundle_All 2.0.0.0 (F335503c) and contao-community-alliance/dc-general dev-develop (93cc9801) -> 2.0.x-dev (doesn't update, I don't know why).

Thanks for help, I can't go further…

**Comment by xantippe** [comment]
same https://github.com/MetaModels/core/issues/977

you can try to update to DCG beta34 and MM alpha10

Please therefore read the following instructions carefully: http://metamodels.readthedocs.org/de/latest/manual/install.html#test-von-speziellen-bundles-via-composer

**Comment by elenadatta** [comment]
I Have the same problem ... I try with 3.5.6 and contao 3.2 with the last MetaModels 2.x (I don't use the composer) ... I'm trying to add a select attribute in a MM but if I select another MM as source table, Id Column, Value Column and Alias column stay empty. Nothing to select inside… Thanks for help

**Comment by JLBA** [comment]
**@xantippe** Ok. 
I'll read carefully and test it  tomorrow. 

For informations, if this can help, I had installed syncCto too. Perhaps did it force update DC-G. I removed it but DC-G now doesn't want to downgrade. 
One year ago, when testing MM, I had problems with MM (1.x ?) and syncCto which didn't like the same version of DC-G (and DC_G à this time).

**Comment by xantippe** [comment]
@JLBA you can try my current files - look at https://community.contao.org/de/showthread.php?55694-Fatal-error-Uncaught-exception-RuntimeException&p=399398&viewfull=1#post399398

**Comment by elenadatta** [comment]
I tried to replace the module metamodelsattribute_select you have posted but my project still doesn't work. I have a dubt ... My server is under php 5.4 ... could this be a problem ? Is php 5.5 required ?

**Comment by xantippe** [comment]
look at https://packagist.org/packages/metamodels/core

5.4 should run...

**Comment by xantippe** [comment]
@JLBA pls try at
https://github.com/MetaModels/attribute_select/blob/master/src/MetaModels/DcGeneral/Events/Table/Attribute/Select/Subscriber.php#L254-L262

```diff
            return
                array
                (
-                    $GLOBALS['TL_LANG']['tl_metamodel_attribute']['select_column_type'] => array_diff_key(
-                        $this->getColumnNamesFromMetaModel($table),
-                        array_flip(array_keys($attributes))
-                    ),
-                    $GLOBALS['TL_LANG']['tl_metamodel_attribute']['select_column_type'] => $attributes
+                    $GLOBALS['TL_LANG']['tl_metamodel_attribute']['select_column_type']['sql'] => array_diff_key(
+                        $this->getColumnNamesFromMetaModel($table),
+                        array_flip(array_keys($attributes))
+                    ),
+                    $GLOBALS['TL_LANG']['tl_metamodel_attribute']['select_column_type']['attribute']=> $attributes

                );
```

**Comment by JLBA** [comment]
Hi Xantippe,
For now, I've downloaded your own files and updated a fresh Contao 3.5.6 + MM 2.0.0 installation. I have  purged all the caches, cleared Composer cache and launched /contao/install.php but no way. 
dc-general is still indicated as 2.0.0-beta33 and doesn't want to update to dev-hotfix/2.0.0-beta34. Below is the answer of Composer.

I'll test other ways you indicate.

```
$ /usr/bin/env php -d memory_limit=1G -d max_execution_time=1500 composer.phar update --no-ansi --no-interaction
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - metamodels/bundle_all 2.0.0 requires metamodels/bundle_attributes ~2.0 -> satisfiable by metamodels/bundle_attributes[2.0.0].
    - metamodels/bundle_all 2.0.0 requires metamodels/bundle_attributes ~2.0 -> satisfiable by metamodels/bundle_attributes[2.0.0].
    - metamodels/bundle_attributes 2.0.0 requires metamodels/attribute_combinedvalues ~2.0 -> satisfiable by metamodels/attribute_combinedvalues[2.0.0-alpha1, 2.1.x-dev, 2.0.x-dev].
    - metamodels/bundle_attributes 2.0.0 requires metamodels/attribute_combinedvalues ~2.0 -> satisfiable by metamodels/attribute_combinedvalues[2.0.0-alpha1, 2.1.x-dev, 2.0.x-dev].
    - metamodels/attribute_combinedvalues 2.0.0-alpha1 requires metamodels/core ~2.0 -> satisfiable by metamodels/core[2.0.0-alpha9, 2.0.0-alpha1, 2.0.0-alpha2, 2.0.0-alpha3, 2.0.0-alpha4, 2.0.0-alpha5, 2.0.0-alpha6, 2.0.0-alpha7, 2.0.0-alpha8, 2.1.x-dev, 2.0.x-dev].
    - metamodels/attribute_combinedvalues 2.0.0-alpha1 requires metamodels/core ~2.0 -> satisfiable by metamodels/core[2.0.0-alpha9, 2.0.0-alpha1, 2.0.0-alpha2, 2.0.0-alpha3, 2.0.0-alpha4, 2.0.0-alpha5, 2.0.0-alpha6, 2.0.0-alpha7, 2.0.0-alpha8, 2.1.x-dev, 2.0.x-dev].
    - metamodels/attribute_combinedvalues 2.1.x-dev requires metamodels/core ~2.0 -> satisfiable by metamodels/core[2.0.0-alpha9, 2.0.0-alpha1, 2.0.0-alpha2, 2.0.0-alpha3, 2.0.0-alpha4, 2.0.0-alpha5, 2.0.0-alpha6, 2.0.0-alpha7, 2.0.0-alpha8, 2.1.x-dev, 2.0.x-dev].
    - metamodels/attribute_combinedvalues 2.0.x-dev requires metamodels/core ~2.0 -> satisfiable by metamodels/core[2.0.0-alpha9, 2.0.0-alpha1, 2.0.0-alpha2, 2.0.0-alpha3, 2.0.0-alpha4, 2.0.0-alpha5, 2.0.0-alpha6, 2.0.0-alpha7, 2.0.0-alpha8, 2.1.x-dev, 2.0.x-dev].
    - metamodels/core 2.0.0-alpha9 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha1 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha2 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha3 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha4 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha5 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha6 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha7 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha8 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.0-alpha9 requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.1.x-dev requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - metamodels/core 2.0.x-dev requires contao-community-alliance/dc-general ~2.0 -> satisfiable by contao-community-alliance/dc-general[2.0.x-dev].
    - Can only install one of: contao-community-alliance/dc-general[dev-hotfix/2.0.0-beta34, 2.0.x-dev].
    - Installation request for contao-community-alliance/dc-general dev-hotfix/2.0.0-beta34 -> satisfiable by contao-community-alliance/dc-general[dev-hotfix/2.0.0-beta34].
    - Installation request for metamodels/bundle_all >=2.0.0.0,<2.1-dev -> satisfiable by metamodels/bundle_all[2.0.0].
```

**Comment by JLBA** [comment]
> look at https://packagist.org/packages/metamodels/core
> 
> 5.4 should run...

What is 5.4. Do you mean Php? I'm under 5.5.23 (in MAMP, I can choose 5.4.39 but is ti a good choice) ?
Which version of metamodels/core do I have to choose : alpha9, x-dev ?

**Comment by xantippe** [comment]
the bug is at the Subscriber.php

you can use the fix https://github.com/MetaModels/attribute_select/issues/48#issuecomment-185084637 (llok at row with +)

**Comment by JLBA** [comment]
Ok. Sorry but what is the path for this file ? I don't find it in my local installation.

**Comment by xantippe** [comment]
ufff! "I hold your hand"

composer\vendor\metamodels\attribute_tags\src\MetaModels\DcGeneral\Events\Table\Attribute\Tags

**Comment by JLBA** [comment]
Sorry, it wasn't clear for me the file to modify was in composer\vendor\metamodels\attribute_tags\src\MetaModels\… Composer is new for me.

**Comment by elenadatta** [comment]
Great!, I modified the subscriber.php file as xantippe wrote ... it works ! Thanks so much

**Comment by xantippe** [comment]
@elenadatta you can use my wishlist http://www.amazon.de/wishlist/26DVT501IVBJY ;-)
alias zonky

**Comment by xantippe** [comment]
Bug is fixed with  dev-hotfix-2.0.0-alpha11 (fff42965)
